{
	"id": "oval:org.opensuse.security:def:202137678",
	"version": "1",
	"class": "vulnerability",
	"metadata": {
		"title": "CVE-2021-37678",
		"affected": {
			"family": "unix",
			"platform": [
				"openSUSE Tumbleweed"
			]
		},
		"reference": [
			{
				"ref_id": "Mitre CVE-2021-37678",
				"ref_url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-37678",
				"source": "CVE"
			},
			{
				"ref_id": "SUSE CVE-2021-37678",
				"ref_url": "https://www.suse.com/security/cve/CVE-2021-37678",
				"source": "SUSE CVE"
			},
			{
				"ref_id": "openSUSE-SU-2022:10014-1",
				"ref_url": "https://lists.opensuse.org/archives/list/security-announce@lists.opensuse.org/thread/U25ZU2T5T4LOLSIYIMGC5PLU4PQQMJE5/",
				"source": "SUSE-SU"
			}
		],
		"description": "\n    TensorFlow is an end-to-end open source platform for machine learning. In affected versions TensorFlow and Keras can be tricked to perform arbitrary code execution when deserializing a Keras model from YAML format. The [implementation](https://github.com/tensorflow/tensorflow/blob/460e000de3a83278fb00b61a16d161b1964f15f4/tensorflow/python/keras/saving/model_config.py#L66-L104) uses `yaml.unsafe_load` which can perform arbitrary code execution on the input. Given that YAML format support requires a significant amount of work, we have removed it for now. We have patched the issue in GitHub commit 23d6383eb6c14084a8fc3bdf164043b974818012. The fix will be included in TensorFlow 2.6.0. We will also cherrypick this commit on TensorFlow 2.5.1, TensorFlow 2.4.3, and TensorFlow 2.3.4, as these are also affected and still in supported range.\n    ",
		"advisory": {
			"from": "security@suse.de",
			"severity": "Important",
			"cve": [
				{
					"text": "CVE-2021-37678 at SUSE",
					"href": "https://www.suse.com/security/cve/CVE-2021-37678/"
				},
				{
					"text": "CVE-2021-37678 at NVD",
					"href": "https://nvd.nist.gov/vuln/detail/CVE-2021-37678",
					"impact": "high",
					"cvss3": "8.8/CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:C/C:H/I:H/A:H"
				}
			],
			"bugzilla": [
				{
					"text": "SUSE bug 1189423",
					"href": "https://bugzilla.suse.com/1189423"
				}
			],
			"issued": {
				"date": "2022-06-30"
			},
			"updated": {
				"date": "2024-09-01"
			},
			"affected_cpe_list": {
				"cpe": [
					"cpe:/o:opensuse:tumbleweed"
				]
			}
		}
	},
	"criteria": {
		"operator": "AND",
		"criterias": [
			{
				"operator": "OR",
				"criterions": [
					{
						"test_ref": "oval:org.opensuse.security:tst:2009925454",
						"comment": "tensorflow-lite-2.9.1-1.1 is installed"
					},
					{
						"test_ref": "oval:org.opensuse.security:tst:2009925455",
						"comment": "tensorflow-lite-devel-2.9.1-1.1 is installed"
					}
				]
			}
		],
		"criterions": [
			{
				"test_ref": "oval:org.opensuse.security:tst:2009634834",
				"comment": "openSUSE Tumbleweed is installed"
			}
		]
	}
}
